x-pack/plugins/logstash/server/routes/pipeline/delete.ts

```diff
@@ -25,10 +24,9 @@ export function registerPipelineDeleteRoute(router: IRouter) {
       router.handleLegacyErrors(async (context, request, response) => {
         const client = context.logstash!.esClient;
 
-        await client.callAsCurrentUser('delete', {
-          index: INDEX_NAMES.PIPELINES,
-          id: request.params.id,
-          refresh: 'wait_for',
+        await client.callAsCurrentUser('transport.request', {
```

Comment:
1. Logstash plugin needs to be migrated to the new Elasticsearch client https://github.com/elastic/elasticsearch-js as we are going to remove the old one in v8.0 Changelog: https://github.com/elastic/kibana/blob/master/src/core/MIGRATION_EXAMPLES.md#elasticsearch-client
2.`transport.request` doesn't provide type safety.
3. The current implementation doesn't prefix endpoints with `_kibana` as outlined in the migration path for 7.x https://github.com/elastic/kibana/issues/81536

In https://github.com/elastic/kibana/issues/82716, we are going to provide a separate Elasticsearch client to address 2 & 3 points. Logstash plugin will migrate to it and adopt all usage places to make sure compatibility with the new Elasticsearch client.

Reply:
Regarding 3, these APIs will not be prefixed with `_kibana` as they exist in their own plugin within Elasticsearch and not within the Kibana system index plugin.

Ideally, I'd like to see us move the Logstash UI to use the system index APIs sooner rather than later to get more of the system index work into users' hands.

Reply:
> Regarding 3, these APIs will not be prefixed with _kibana as they exist in their own plugin within Elasticsearch and not within the Kibana system index plugin.

Makes sense to merge it then. You just need to address 1st point then. Can be done in the follow-up.

Hopefully we can fix the 2nd problem as well with https://github.com/elastic/kibana/pull/80405#discussion_r521212486

Reply:
I will address 1st point as a follow-up issue

Reply:
@restrry is there a meta issue/transition plan to support the new elasticsearc-js client library?

Reply:
@roaksoax https://github.com/elastic/kibana/issues/83910